server/index-node/src/resolver.rs

```diff
@@ -34,19 +35,41 @@ impl TryFromValue for DeploymentAssignment {
     }
 }
 
+#[derive(Debug)]
+struct ProofOfIndexingInput {
+    // The id of the subgraph.
+    subgraph_name: String,
```

Comment:
This should really be the ID (i.e. the `Qm...` hash of the subgraph deployment), not the subgraph name. The name is a moving target, whereas the ID always points to the same subgraph data.